sdk/ai/ai-inference-rest/samples/v1-beta/javascript/streamingToolCall.js

```diff
@@ -0,0 +1,181 @@
+  },
+};
+
+const getWeatherFunc = (location, unit) => {
+  if (unit != "celsius") {
+    unit = "fahrenheit";
+  }
+  return `The temperature in ${location} is 72 degrees ${unit}`;
+};
+
+const updateToolCalls = (toolCallArray, functionArray) => {
+  const dummyFunction = { name: "", arguments: "", id: "" };
+  while (functionArray.length < toolCallArray.length) {
+    functionArray.push(dummyFunction);
+  }
+
+  let index = 0;
+  for (const toolCall of toolCallArray) {
+    if (toolCall.function.name) {
+      functionArray[index].name = toolCall.function.name;
+    }
+    if (toolCall.id) {
+      functionArray[index].id = toolCall.id;
+    }
+    if (toolCall.function.arguments) {
+      functionArray[index].arguments += toolCall.function.arguments;
+    }
+    index++;
+  }
+};
+
+const handleToolCalls = (functionArray) => {
+  const messageArray = [];
+  for (const func of functionArray) {
+    const funcArgs = JSON.parse(func.arguments);
+    let content = "";
+
+    switch (func.name) {
+      case "get_current_weather":
+        content = getWeatherFunc(funcArgs.location, funcArgs.unit ?? "fahrenheit");
+        messageArray.push({
+          role: "tool",
+          content,
+          tool_call_id: func.id,
+          name: func.name,
+        });
+        break;
+
+      default:
+        console.log(`unknown function ${func.name}`);
+        break;
+    }
+  }
+  return messageArray;
+};
+
+const streamToString = async (stream) => {
+  // lets have a ReadableStream as a stream variable
+  const chunks = [];
+
+  for await (const chunk of stream) {
+    chunks.push(Buffer.from(chunk));
+  }
+
+  return Buffer.concat(chunks).toString("utf-8");
+};
+
+async function main() {
+  const credential = new DefaultAzureCredential();
+  // auth scope for AOAI resources is currently https://cognitiveservices.azure.com/.default
+  // (only needed when targetting AOAI, do not use for Serverless API or Managed Computer Endpoints)
+  const scopes = ["https://cognitiveservices.azure.com/.default"];
+  const clientOptions = { credentials: { scopes } };
+  const client = ModelClient(modelEndpoint, credential, clientOptions);
+
+  const messages = [{ role: "user", content: "What's the weather like in Boston?" }];
+
+  let toolCallAnswer = "";
+  let awaitingToolCallAnswer = true;
+  while (awaitingToolCallAnswer) {
+    const response = await client
+      .path("/chat/completions")
+      .post({
+        body: {
+          messages,
+          tools: [
+            {
+              type: "function",
+              function: getCurrentWeather,
+            },
+          ],
+          stream: true,
+        },
+      })
+      .asNodeStream();
+
+    const stream = response.body;
+    if (!stream) {
+      throw new Error("The response stream is undefined");
+    }
+
+    if (response.status !== "200") {
+      throw new Error(`Failed to get chat completions: ${await streamToString(stream)}`);
+    }
+
+    const sses = createSseStream(stream);
```

Comment:
Would it be possible to get SSE events as result from chat completion?

Reply:
> Would it be possible to get SSE events as result from chat completion?

Not sure how that would work, could you sketch out what you mean a bit more?